src/network/networkd-wwan-bus.c

```diff
@@ -0,0 +1,362 @@
+                { "Interface",  "s",     map_name,       offsetof(Bearer, name)      },
+                { "Connected",  "b",     NULL,           offsetof(Bearer, connected) },
+                { "Ip4Config",  "a{sv}", map_ip4_config, 0,                          },
+                { "Ip6Config",  "a{sv}", map_ip6_config, 0,                          },
+                { "Properties", "a{sv}", map_properties, 0,                          },
+                {}
+        };
+
+        Bearer *b = ASSERT_PTR(userdata);
+        const sd_bus_error *e;
+        int r;
+
+        assert(message);
+
+        b->slot = sd_bus_slot_unref(b->slot);
+
+        e = sd_bus_message_get_error(message);
+        if (e) {
+                bool removed = false;
+
+                if (sd_bus_error_has_name(e, SD_BUS_ERROR_UNKNOWN_METHOD))
+                        /* The path is already removed? */
+                        removed = true;
+
+                r = sd_bus_error_get_errno(e);
+                log_full_errno(removed ? LOG_DEBUG : LOG_WARNING, r,
+                               "Could not get properties of bearer \"%s\": %s",
+                               b->path, bus_error_message(e, r));
+
+                bearer_drop(b);
+                return 0;
+        }
+
+        r = bus_message_map_all_properties(message, map, BUS_MAP_BOOLEAN_AS_BOOL, ret_error, b);
+        if (r < 0)
+                return log_warning_errno(r, "Failed to parse properties of bearer \"%s\": %s", b->path, bus_error_message(ret_error, r));
+
+        (void) bearer_update_link(b);
+        return 0;
+}
+
+static int bearer_initialize(Bearer *b) {
+        int r;
+
+        assert(b);
+        assert(b->manager);
+        assert(sd_bus_is_ready(b->manager->bus) > 0);
+        assert(b->path);
+
+        b->slot = sd_bus_slot_unref(b->slot);
+
+        r = sd_bus_call_method_async(
+                        b->manager->bus,
+                        &b->slot,
+                        "org.freedesktop.ModemManager1",
+                        b->path,
+                        "org.freedesktop.DBus.Properties",
+                        "GetAll",
+                        bearer_get_all_handler,
+                        b,
+                        NULL);
+        if (r < 0)
+                return log_warning_errno(r, "Could not get properties of bearer \"%s\": %m", b->path);
+
+        return 0;
+}
+
+static int bearer_new_and_initialize(Manager *manager, const char *path) {
+        _cleanup_(bearer_freep) Bearer *b = NULL;
+        int r;
+
+        assert(manager);
+        assert(path);
+
+        r = bearer_new(manager, path, &b);
+        if (r < 0)
+                return log_warning_errno(r, "Failed to allocate new bearer \"%s\": %m", path);
+
+        r = bearer_initialize(b);
+        if (r < 0)
+                return r;
+
+        TAKE_PTR(b);
+        return 0;
+}
+
+static int on_path(const char *path, void *userdata) {
```

Comment:
`on_path` is not a very good name. And I think it's be nicer to do the cast separately:
```C
Set *set = ASSERT_PTR(userdata);
return set_put_strdup(set, path);
```